plugins/bufferlines/barbecue.nix

```diff
@@ -0,0 +1,190 @@
+        '';
+
+      showNavic = helpers.defaultNullOpts.mkBool true ''
+        Whether to show/use navic in the winbar.
+      '';
+
+      leadCustomSection =
+        helpers.defaultNullOpts.mkStr
+        ''
+          function()
+            return " "
+          end
+        ''
+        ''
+          Get leading custom section contents.
+          NOTE: This function shouldn't do any expensive actions as it is run on each render.
+        '';
+
+      customSection =
+        helpers.defaultNullOpts.mkStr
+        ''
+          function()
+            return " "
+          end
+        ''
+        ''
+          Get custom section contents.
+          NOTE: This function shouldn't do any expensive actions as it is run on each render.
+        '';
+
+      theme = helpers.defaultNullOpts.mkStr "auto" ''
+        Theme to be used for generating highlight groups dynamically.
+      '';
+
+      contextFollowIconColor = helpers.defaultNullOpts.mkBool false ''
+        Whether context text should follow its icon's color.
+      '';
+
+      symbols = {
+        modified = helpers.defaultNullOpts.mkStr "●" ''
+          Modification indicator.
+        '';
+
+        ellipsis = helpers.defaultNullOpts.mkStr "…" ''
+          Truncation indicator.
+        '';
+
+        separator = helpers.defaultNullOpts.mkStr "" ''
+          Entry separator.
+        '';
+      };
+
+      kinds =
+        mapAttrs
+        (
+          name: default:
+            helpers.defaultNullOpts.mkStr default "icon for ${name}."
+        )
+        {
+          File = "";
+          Module = "";
+          Namespace = "";
+          Package = "";
+          Class = "";
+          Method = "";
+          Property = "";
+          Field = "";
+          Constructor = "";
+          Enum = "";
+          Interface = "";
+          Function = "";
+          Variable = "";
+          Constant = "";
+          String = "";
+          Number = "";
+          Boolean = "";
+          Array = "";
+          Object = "";
+          Key = "";
+          Null = "";
+          EnumMember = "";
+          Struct = "";
+          Event = "";
+          Operator = "";
+          TypeParameter = "";
+        };
+    };
+
+  config = let
+    setupOptions = with cfg;
+      {
+        inherit
+          modifiers
+          modified
+          theme
+          symbols
+          kinds
+          ;
+        attach_navic = attachNavic;
+        create_autocmd = createAutocmd;
+        include_buftypes = includeBuftypes;
+        exclude_filetypes = excludeFiletypes;
+        show_dirname = showDirname;
+        show_basename = showBasename;
+        show_modified = showModified;
+        show_navic = showNavic;
+        lead_custom_section = leadCustomSection;
+        custom_section = customSection;
+        context_follow_icon_color = contextFollowIconColor;
+      }
+      // cfg.extraOptions;
```

Comment:
```suggestion
        attach_navic = attachNavic;
        create_autocmd = createAutocmd;
        include_buftypes = includeBuftypes;
        exclude_filetypes = excludeFiletypes;
        modifiers = {
          inherit (modifiers)
            dirname
            basename
            ;
        };
        show_dirname = showDirname;
        show_basename = showBasename;
        show_modified = showModified;
        modified = helpers.ifNonNull' modified; (helpers.mkRaw modified);
        show_navic = showNavic;
        lead_custom_section = helpers.ifNonNull' leadCustomSection (helpers.mkRaw leadCustomSection);
        custom_section = helpers.ifNonNull' customSection (helpers.mkRaw customSection);
        inherit theme;
        context_follow_icon_color = contextFollowIconColor;
        symbols = {
          inherit (symbols);
            modified
            ellipsis
            separator
            ;
        };
        inherit kinds;       
      }
      // cfg.extraOptions;
```

Reply:
1. You had forgotten to convert (`helpers.mkRaw`) the values of the options that expect lua code
2. Keep the same order in `setupOptions` as the one used when declaring options.

Reply:
1. yeh I was waiting to confirm above if 1 was the right approach!
2. ahh yes the ordering good shout